init-tools.sh

```diff
@@ -15,6 +15,43 @@ __PROJECT_JSON_FILE=$__PROJECT_JSON_PATH/project.json
 __PROJECT_JSON_CONTENTS="{ \"dependencies\": { \"Microsoft.DotNet.BuildTools\": \"$__BUILD_TOOLS_PACKAGE_VERSION\" }, \"frameworks\": { \"netcoreapp1.0\": { } } }"
 __INIT_TOOLS_DONE_MARKER=$__PROJECT_JSON_PATH/done
 
+# Modified platform detection logic from dotnet/cli/scripts/obtain/dotnet-install.sh 16692fc
```

Comment:
The issue with the current flow is that we use `dotnet restore` to get the buildtools package, so we need the CLI first. We could change that--download buildtools directly and run a script in there to get CLI--but https://github.com/dotnet/buildtools/issues/558 is a plan to reduce the init-tools.cmd/sh here to just get another init script from the buildtools github repo. I think that would reduce the per-repo script to its bare minimum.

To share even more, the buildtools init script could even redirect to the CLI repo's install script. We could hook that up whenever we want because with that plan, it's easy to push through changes to the init script.